### PR TITLE
Some TS refactoring & Spanish and Catalan translations

### PR DESCRIPTION
**Pull request by TheDavidDelta**
It's the first time I contribute to a public repository so I'll try to do this well.

+ Firstly, I've refactored some TypeScript codes (translation_report, i18next) with ECMAScript functional paradigma rules for a small improvement on the cleanliness.
+ Secondly, I've finished the Spanish translation file.
+ And lastly, I've created a Catalan translation file, completed it to 100% and added it to translation_report and i18next, executing satisfactorily the refactored translation_report script that modified the README.

Hope you like my contribution!

### CODE REVIEW

File: ui/src/i18next.ts [hunk 1]
```diff
@@ -25,12 +26,10 @@ const resources = {
...
   it,
   fi,
+  ca,
 };
 
-function format(value: any, format: any, lng: any) {
-  if (format === 'uppercase') return value.toUpperCase();
-  return value;
-}
+const format = (value: any, format: any, lng: any) => format === 'uppercase' ? value.toUpperCase() : value;
```
Comment (dessalines) [inline ui/src/i18next.ts]: Could you revert this, I can't stand lambda functions. The conditional ternary is good tho.

File: ui/src/translations/ca.ts [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+      user_already_exists: "L'usuari ja existeix.",
+      email_already_exists: 'El correu ja és en ús.',
+      couldnt_update_user: "No s'ha pogut actualitzar l'usuari.",
+      system_err_login: 'Error del sistema. Intenti tancar sessió i ingressar de nou.',
+      couldnt_create_private_message: "No s'ha pogut crear el missatge privat.",
+      no_private_message_edit_allowed: 'Sense permisos per a editar el missatge privat.',
+      couldnt_update_private_message: "No s'ha pogut actualitzar el missatge privat.",
+    },
+  };
+
```
Comment (dessalines) [inline ui/src/translations/ca.ts]: Sweet, thanks!

File: ui/src/translations/es.ts [hunk 3]
```diff
@@ -192,8 +223,11 @@ export const es = {
...
+    email_already_exists: 'El correo ya está en uso.',
     couldnt_update_user: 'No se pudo actualizar el usuario.',
-    system_err_login:
-      'Error del sistema. Intente cerrar sesión e ingresar de nuevo.',
+    system_err_login: 'Error del sistema. Intente cerrar sesión e ingresar de nuevo.',
+    couldnt_create_private_message: "No se pudo crear el mensaje privado.",
+    no_private_message_edit_allowed: 'Sin permisos para editar el mensaje privado.',
+    couldnt_update_private_message: "No se pudo actualizar el mensaje privado.",
   },
 };
```
Comment (dessalines) [inline ui/src/translations/es.ts]: Thanks for adding these btw.

File: ui/src/version.ts [hunk 4]
```diff
@@ -1 +1 @@
-export let version: string = 'v0.6.5';
+export const version: string = 'v0.6.5';
```
Comment (dessalines) [inline ui/src/version.ts]: This is generated from my `deploy.sh` script, but don't worry, I can change that one.

File: ui/translation_report.ts [hunk 5]
```diff
@@ -9,52 +9,46 @@ import { ru } from './src/translations/ru';
...
+
+const open = '<!-- translations -->';
+const close = '<!-- translationsstop -->';
+
+const readmeTxt = fs.readFileSync(readmePath, { encoding: 'utf8' });
+
+const before = readmeTxt.split(open)[0];
+const after = readmeTxt.split(close)[1];
+
+const difference = (a: Array<string>, b: Array<string>): Array<string> => a.filter(x => !b.includes(x));
```
Comment (dessalines) [inline ui/translation_report.ts]: Again, no lambda functions.

**Comment by TheDavidDelta** [comment]
How should I revert lambdas? Change code and make another pull request or can it be appended to this one? I've never done a pull request.

**Comment by dessalines** [comment]
Just do another commit to your master branch and it'll add it automatically.

**Comment by dessalines** [comment]
Thanks for this again!

**Comment by StaticallyTypedRice** [comment]
Really glad to see more translations!

**Pull request merged**